src/history/hs_cursor.c

```diff
@@ -246,52 +243,12 @@ __hs_find_upd_int(WT_SESSION_IMPL *session, uint32_t btree_id, WT_ITEM *key,
     if (read_timestamp == WT_TS_NONE)
         read_timestamp = WT_TS_MAX;
 
-    WT_ERR_NOTFOUND_OK(
-      __wt_hs_cursor_position(session, hs_cursor, btree_id, key, read_timestamp, NULL), true);
+    hs_cursor->set_key(hs_cursor, 4, btree_id, key, read_timestamp, UINT64_MAX);
+    WT_ERR_NOTFOUND_OK(hs_cursor->search_near(hs_cursor, &cmp), true);
```

Comment:
This is not correct yet. I'm waiting on the helper functions to test this. @hbokhari